components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -377,47 +396,96 @@ func (g *rtkSerial) closePort() {
 	}
 }
 
-// receiveAndWriteSerial connects to NTRIP receiver and sends correction stream to the MovementSensor through serial.
-func (g *rtkSerial) receiveAndWriteSerial() {
-	defer g.activeBackgroundWorkers.Done()
+// connectAndParseSourceTable connects to the NTRIP caster, gets and parses source table
+// from the caster.
+func (g *rtkSerial) connectAndParseSourceTable() error {
```

Comment:
Too many connect functions. I am running out of ideas for names. Let me know if this name makes any sense for this function.